cmd/bucket-metadata-sys.go

```diff
@@ -177,6 +178,40 @@ func (sys *BucketMetadataSys) save(ctx context.Context, meta BucketMetadata) err
 // Delete delete the bucket metadata for the specified bucket.
 // must be used by all callers instead of using Update() with nil configData.
 func (sys *BucketMetadataSys) Delete(ctx context.Context, bucket string, configFile string) (updatedAt time.Time, err error) {
+	if configFile == bucketLifecycleConfig {
+		// Get bucket config from current site
+		meta, e := globalBucketMetadataSys.GetConfigFromDisk(ctx, bucket)
+		if e != nil && !errors.Is(e, errConfigNotFound) {
+			return updatedAt, e
+		}
+		var expiryRuleRemoved bool
+		if len(meta.LifecycleConfigXML) > 0 {
+			var lcCfg lifecycle.Lifecycle
+			if err := xml.Unmarshal(meta.LifecycleConfigXML, &lcCfg); err != nil {
+				return updatedAt, err
+			}
+			// find a single expiry rule set the flag
+			for _, rl := range lcCfg.Rules {
+				if !rl.Expiration.IsNull() {
+					expiryRuleRemoved = true
+					break
+				}
+			}
+		}
+
+		// Form empty ILM details with `ExpiryUpdatedAt` field and save
+		var cfgData []byte
+		if expiryRuleRemoved {
+			var lcCfg lifecycle.Lifecycle
+			currtime := time.Now()
+			lcCfg.ExpiryUpdatedAt = &currtime
+			cfgData, err = xml.Marshal(lcCfg)
+			if err != nil {
+				return updatedAt, err
+			}
+		}
+		return sys.updateAndParse(ctx, bucket, configFile, cfgData, false)
```

Comment:
why do you need this change? expiryUpdatedAt timestamp is already preserved in metadata for lifecycle changes. You will just need to return it in GetLifecycleConfig when config is nil

Reply:
Ok, let me try that.

Reply:
This is required because in normal delete it sets `nil` for config file. In case of lifecycle config and delete, we still want to retain the latest `ExpiryUpdatedAt` value with no rules, so that while sync it would figure out on other peers that peer with latest `ExpiryUpdatedAt` has no rules (i.e. deleted) and we need to delete as well. 
Basically for replicating the removal of expiry rules across sites.
Without this, other peers would find their `ExpiryUpdatedAt` value as bigger than the site where this delete actually happened, and they would replicate the rules back.